File: Serene/Serene.Script/Membership/Authorization.cs

namespace Serene
{
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    public static class Authorization
    {
        [IntrinsicProperty]
        public static ScriptUserDefinition UserDefinition { get; private set; }

        public static bool HasPermission(string permissionKey)
        {
            return false;
        }
    }
}

File: Serene/Serene.Web/Modules/Administration/User/Authentication/Authorization.ts
namespace Serene.Authorization {
    export declare let userDefinition: ScriptUserDefinition;

    Object.defineProperty(Authorization, 'userDefinition', {
        get: function () {
            return Q.getRemoteData('UserData');
        }
    });

    export function hasPermission(permissionKey: string) {
        return Q.Authorization.hasPermission(permissionKey);
    }
}
